manifests/repo.pp

```diff
@@ -74,19 +76,18 @@
           $repo_path   = 'mongodb-org'
         }
 
-        $mongover = split($version, '[.]')
         $location = $facts['os']['name'] ? {
           'Debian' => "https://${repo_domain}/apt/debian",
           'Ubuntu' => "https://${repo_domain}/apt/ubuntu",
           default  => undef
         }
-        $release     = "${facts['os']['distro']['codename']}/${repo_path}/${mongover[0]}.${mongover[1]}"
+        $release     = "${facts['os']['distro']['codename']}/${repo_path}/${version}"
         $repos       = $facts['os']['name'] ? {
           'Debian' => 'main',
           'Ubuntu' => 'multiverse',
           default => undef
         }
-        $key = "${mongover[0]}.${mongover[1]}" ? {
+        $key = $version ? {
           '5.0'   => 'F5679A222C647C87527C2F8CB00A0BD1E2C63C11',
           '4.4'   => '20691EEC35216C63CAF66CE1656408E390CFB1F5',
           '4.2'   => 'E162F504A20CDF15827F718D4B7C549A058F8B6B',
```

Comment:
If you are going to remove support for old repos, might as well remove the old keys from the code here.

Reply:
I don't really know what version the default key belonged to.
Changed it to undef, but if someone hits this they'll get
`Evaluation Error: Error while evaluating a Function Call, key hash must contain a key name (for apt::keyring) or an id (for apt::key) (file: /etc/puppetlabs/code/environments/production/modules/apt/manifests/source.pp, line: 143, column: 9`

At the moment I guess we should require repo_version < 6.0 as well ..

Reply:
May be we should change the $repo_version from String to Enum['4.4','5.0']?

Reply:
The default key was from version 3.0, according to the Git history.

Not sure about the enum, that would make support for 6.0 a just little bit harder.

Reply:
Well, this class can't support 6.0 without adding the 6.0 key as well can it?

Reply:
I'd just change the default key to the one for 4.4, as it makes no sense to have that old one.

Reply:
> Well, this class can't support 6.0 without adding the 6.0 key as well can it?

The repo is not required, think Redhat Satellite.